Core.DomainModel/Organization/Organization.cs

```diff
@@ -279,5 +279,27 @@ public Maybe<OperationError> ConnectToExternalOrganizationHierarchy(Organization
                     throw new ArgumentOutOfRangeException();
             }
         }
+
+        public Result<OrganizationUnit, OperationError> RemoveOrganizationUnit(int unitId)
+        {
+            var unit = OrgUnits.FirstOrDefault(x => x.Id == unitId);
+            if (unit == null)
+                return new OperationError("Unit doesn't exist in the organization", OperationFailure.NotFound);
+            if (unit.Origin != OrganizationUnitOrigin.Kitos)
+                return new OperationError("Only a KITOS unit can be deleted", OperationFailure.BadState);
+            if (unit.IsUsed())
+                return new OperationError("Unit is being used", OperationFailure.BadState);
+
+            // attach children to parent of this instance to avoid orphans
+            // parent id will never be null because users aren't allowed to delete the root node
```

Comment:
snatch the new implementation from my PR because this method was one of the brittle ones